AU2/database/model/Assassin.py

```diff
@@ -42,9 +58,55 @@ def __post_init__(self):
             self.identifier = f"{self.real_name} ({self.pseudonyms[0][:15]}{dotdotdot}) ID: {self._secret_id}"
 
     def get_pseudonym(self, i: int) -> str:
-        if i >= len(self.pseudonyms):
-            return self.pseudonyms[-1]
+        if i >= len(self.pseudonyms) or not self.pseudonyms[i]:
+            # fallback to last non-null pseudonym

```

Comment:
Yep; although actually deletion sets them to `""` rather than `None`, because the latter gets JSONified as the *string* `"None"` rather than an actual null value... (So I should probably clarify this in the comment as well!)